### PR TITLE
endpoint and API for listing projects for a given secret on google

### PR DESCRIPTION
**Pull request by lpuskas**
- new endpoint to retrieve google projects visible for a given secret
- extended the openapi spec with the new call
- generated client
- made project field optional when creating google secrets

relates to #1092

### CODE REVIEW

File: api/projects.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	"github.com/banzaicloud/pipeline/internal/platform/gin/utils"
+	"github.com/banzaicloud/pipeline/pkg/providers"
+	"github.com/banzaicloud/pipeline/secret/verify"
+	"github.com/gin-gonic/gin"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/cloudresourcemanager/v1"
+)
+
+type ListProjectsResponse struct {
+	Projects []*cloudresourcemanager.Project
```
Comment (bonifaido) [inline api/projects.go]: Shouldn't be this tagged with `json` tags?

File: client/api/openapi.yaml [hunk 2]
```diff
@@ -8289,6 +8337,41 @@ components:
         vulnerabilities:
           $ref: '#/components/schemas/VulnerabilityList'
       type: object
+    ListProjectsResponse:
+      description: wraps a list of project details information
+      example:
+        Projects:
```
Comment (bonifaido) [inline client/api/openapi.yaml]: should be `projects` after the struct tagging.

File: api/projects.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+		return nil
+	}); err != nil {
+		log.WithError(err).Error("could not retrieve projects")
+		ginutils.ReplyWithErrorResponse(c, errorResponseFrom(err))
+	}
+
+}
+
+// HttpClient builds a http client with the service account available through the secret and organization
+func (sc *servicesContext) HttpClient() (*http.Client, error) {
```
Comment (stoader) [inline api/projects.go]: Since this is not used from other packages probably shouldn't be public.

File: api/projects.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+	}
+	log.Debugf("retrieving projects for orgId: [%d], secretId [%s]",
+		organization.ID, secretID)
+	servicesCtx := newServicesCtx(organization.ID, secretID)
+
+	cli, err := servicesCtx.HttpClient()
+	if err != nil {
+		log.WithError(err).Error("could not build http client")
+		ginutils.ReplyWithErrorResponse(c, errorResponseFrom(err))
+	}
```
Comment (colin014) [inline api/projects.go]: I think a return is missing here

File: api/projects.go [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+	projectsSvc, err := servicesCtx.ProjectsService(cli)
+	if err != nil {
+		log.WithError(err).Error("could not build projects service")
+		ginutils.ReplyWithErrorResponse(c, errorResponseFrom(err))
```
Comment (colin014) [inline api/projects.go]: Same as above

**Pull request merged**